src/aind_data_access_api/helpers/data_schema.py

```diff
@@ -74,3 +78,111 @@ def validate_qc(qc_data: dict, allow_invalid: bool = False):
             return qc_data
         else:
             raise e
+
+
+def get_quality_control_by_names(
+    client: MetadataDbClient,
+    names: List[str],
+    allow_invalid: bool = False,
+) -> List[Union[QualityControl, dict]]:
+    """Using a connected DocumentDB client, retrieve the QualityControl object
+    for a list of records.
+
+    Parameters
+    ----------
+    client : MetadataDbClient
+        A connected DocumentDB client.
+    names : List[str],
+        name fields in DocDB
+    allow_invalid : bool, optional
+        return invalid QualityControl as dict if True, by default False
+    """
+    records = fetch_records_by_filter_list(
+        client,
+        filter_key="name",
+        filter_values=names,
+        projection={"quality_control": 1},
+    )
+
+    qcs = [
+        validate_qc(record["quality_control"], allow_invalid=allow_invalid)
+        for record in records
+    ]
+
+    return qcs
+
+
+def get_quality_control_status_df(
+    client: MetadataDbClient,
+    names: List[str],
+    date: Optional[datetime] = None,
+) -> pd.DataFrame:
+    """Using a connected DocumentDB client, retrieve the status of all
+    QualityControl objects for a list of records.
+
+    Parameters
+    ----------
+    client : MetadataDbClient
+        A connected DocumentDB client.
+    names : List[str],
+        name fields in DocDB
+    """
+    if date is None:
+        date = datetime.now(tz=timezone.utc)
+
+    qcs = [
+        get_quality_control_by_name(client, name=name, allow_invalid=False)
+        for name in names
+    ]
```

Comment:
should use new method
```suggestion
    qcs = get_quality_control_by_names(
        client=client, names=names, allow_invalid=False
    )
```